backend/ohq/views.py

```diff
@@ -222,19 +228,28 @@ def list(self, request, *args, **kwargs):
         return super().list(request, *args, **kwargs)
 
 
-class QuestionSearchView(generics.ListAPIView):
+class QuestionSearchView(XLSXFileMixin, generics.ListAPIView):
     filter_backends = [DjangoFilterBackend]
     filterset_class = QuestionSearchFilter
     pagination_class = QuestionSearchPagination
     permission_classes = [QuestionSearchPermission | IsSuperuser]
+    renderer_classes = [CamelCaseJSONRenderer, CamelCaseBrowsableAPIRenderer, XLSXRenderer]
     serializer_class = QuestionSerializer
+    filename = "questions.xlsx"
 
     def get_queryset(self):
         qs = Question.objects.filter(
             queue__in=Queue.objects.filter(course=self.kwargs["course_pk"])
         ).order_by("time_asked")
         return prefetch(qs, self.serializer_class)
 
+    @property
+    def paginator(self):
+        if isinstance(self.request.accepted_renderer, XLSXRenderer):  # xlsx download
```

Comment:
This seems fine, I'd just leave a more descriptive comment -- maybe link to documentation on `accepted_renderer` and how it's set based on the request?

Reply:
Makes sense. Added

Reply:
Do we want to be slightly more general? `if self.request.accepted_handler.format == 'xlsx'`? I don't think in practice there's a difference, but it'd be more "forwards compatible" in case we change something.